backend/sim/channel/asset.go

```diff
@@ -40,17 +43,17 @@ func NewRandomAsset(rng *rand.Rand) *Asset {
 
 // MarshalBinary marshals the address into its binary representation.
 func (a Asset) MarshalBinary() ([]byte, error) {
-	buff := bytes.NewBuffer(make([]byte, 0, AssetBinaryLen))
-	if err := perunio.Encode(buff, a.ID); err != nil {
-		return nil, err
-	}
-	return buff.Bytes(), nil
+	data := make([]byte, assetLen)
+	byteOrder.PutUint64(data, uint64(a.ID))
+	return data, nil
 }
 
 // UnmarshalBinary unmarshals the asset from its binary representation.
 func (a *Asset) UnmarshalBinary(data []byte) error {
-	if len(data) != AssetBinaryLen {
-		return fmt.Errorf("unexpected address length %d, want %d", len(data), AssetBinaryLen) //nolint: goerr113
+	if len(data) != assetLen {
+		//nolint: goerr113  // We do not want to define this as constant error.
+		return fmt.Errorf("unexpected length %d, want %d", len(data), assetLen)
```

Comment:
`//nolint` should be used on the line where it is relevant. (Also typically no space between '//nolint:' and the specified linter name, see [here](https://golangci-lint.run/usage/false-positives/).)
```suggestion
		return fmt.Errorf("unexpected length %d, want %d", len(data), assetLen) //nolint:goerr113  // We do not want to define this as constant error.
```

Reply:
I had put it above the line, because it was making the line very long. But, I am okay with either of the approaches.